### PR TITLE
Use recommended Node and Rust gitignore files

### PR DESCRIPTION
**Pull request by Turbo87**
see https://github.com/github/gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,135 @@
...
+### Rust template
+
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```
Comment (ef4) [inline .gitignore]: I think we should remove Cargo.lock from this list because we're shipping an executable, not a library.

Comment (rwjblue) [reply .gitignore]: From this article:

> If you’re building a non-end product, such as a rust library that other rust packages will depend on, put Cargo.lock in your .gitignore. If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of staticlib or cdylib, check Cargo.lock into git. If you're curious about why that is, see ["Why do binaries have Cargo.lock in version control, but not libraries?" in the FAQ](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries).

Comment (Turbo87) [reply .gitignore]: you are totally right. I missed that when I clicked on "Generate"... 😅 

should be fixed now :)

Comment (Turbo87) [reply .gitignore]: FWIW I usually even commit the `Cargo.lock` file for libraries, similar to how we do it with `yarn.lock` in Ember addons.

Comment (ef4) [reply .gitignore]: Agreed. It's nice to have a lock to keep your own development sane.

**Pull request merged**